core/src/test/scala/org/broadinstitute/dsde/rawls/billing/GoogleBillingProjectLifecycleSpec.scala

```diff
@@ -132,4 +165,126 @@ class GoogleBillingProjectLifecycleSpec extends AnyFlatSpec {
+        ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
+      )
+    ).thenReturn(Future.successful(Map(WorkbenchEmail(userInfo.userEmail.value) -> Seq())))
+
+    when(repo.setBillingProfileId(createRequest.projectName, profileModel.getId))
+      .thenReturn(Future.successful(1))
+
+    doReturn(Future.successful())
+      .when(wsmResourceRecordDao)
+      .create(any)
+
+    Await.result(bp.postCreationSteps(
+                   createRequest,
+                   mock[MultiCloudWorkspaceConfig],
+                   testContext
+                 ),
+                 Duration.Inf
+    )
+
+    verify(repo, Mockito.times(1)).setBillingProfileId(
+      createRequest.projectName,
+      profileModel.getId
+    )
+  }
+
+  it should "add additional members to the BPM policy during billing project creation if specified" in {
+    val repo = mock[BillingRepository]
+    val bpm = mock[BillingProfileManagerDAO]
+    val samDAO = mock[SamDAO]
+    val wsmResourceRecordDao = mock[WorkspaceManagerResourceMonitorRecordDao]
+    val bp = new GoogleBillingProjectLifecycle(repo, bpm, samDAO, mock[GoogleServicesDAO])
+
+    val user1Email = "user1@foo.bar"
+    val user2Email = "user2@foo.bar"
+    val user3Email = "user3@foo.bar"
+
+    val createRequestWithMembers = createRequest.copy(members =
+      Some(
+        Set(
+          ProjectAccessUpdate(user1Email, ProjectRoles.Owner),
+          ProjectAccessUpdate(user2Email, ProjectRoles.Owner),
+          ProjectAccessUpdate(user3Email, ProjectRoles.User)
+        )
+      )
+    )
+
+    when(
+      bpm.createBillingProfile(
+        ArgumentMatchers.eq(createRequestWithMembers.projectName.value),
+        ArgumentMatchers.eq(createRequestWithMembers.billingInfo),
+        any(),
+        ArgumentMatchers.eq(testContext)
+      )
+    )
+      .thenReturn(profileModel)
+
+    when(
+      samDAO.syncPolicyToGoogle(
+        ArgumentMatchers.eq(SamResourceTypeNames.billingProject),
+        ArgumentMatchers.eq(createRequest.projectName.value),
+        ArgumentMatchers.eq(SamBillingProjectPolicyNames.owner)
+      )
+    ).thenReturn(Future.successful(Map(WorkbenchEmail(userInfo.userEmail.value) -> Seq())))
+
+    when(repo.setBillingProfileId(createRequestWithMembers.projectName, profileModel.getId))
+      .thenReturn(Future.successful(1))
+
+    doReturn(Future.successful())
+      .when(wsmResourceRecordDao)
+      .create(any)
+
+    Await.result(bp.postCreationSteps(
+                   createRequestWithMembers,
+                   mock[MultiCloudWorkspaceConfig],
+                   testContext
+                 ),
+                 Duration.Inf
+    )
+
+    verify(bpm, Mockito.times(2)).addProfilePolicyMember(
+      ArgumentMatchers.eq(profileModel.getId),
+      ArgumentMatchers.eq(ProfilePolicy.Owner),
+      ArgumentMatchers.argThat(arg => Set(user1Email, user2Email).contains(arg)),
+      any[RawlsRequestContext]
+    )
```

Comment:
I'm a bit confused by this verification -- the emails specifically.  This would pass if `user1Email` was added to the policy twice and `user2Email` was never added, is that intended?

Reply:
I just copied this test case from the Azure spec file and didn't examine it too closely. I will see about improving it.

Reply:
That would be great, it would be good to understand the expected behavior here.

Reply:
I updated the test and the place I copied it from in the last commit.